mp2-v1/src/receipt_extraction/public_inputs.rs

```diff
@@ -0,0 +1,170 @@
+//! Public inputs for Receipt Extraction circuits
+
+use mp2_common::{
+    array::Array,
+    keccak::{OutputHash, PACKED_HASH_LEN},
+    mpt_sequential::ReceiptKeyWire,
+    public_inputs::{PublicInputCommon, PublicInputRange},
+    types::{CBuilder, GFp, GFp5, CURVE_TARGET_LEN},
+    utils::{convert_point_to_curve_target, convert_slice_to_curve_point, FromTargets},
+};
+
+use plonky2::{
+    field::{extension::FieldExtension, types::Field},
+    iop::target::Target,
+};
+use plonky2_ecgfp5::{
+    curve::curve::WeierstrassPoint,
+    gadgets::curve::{CircuitBuilderEcGFp5, CurveTarget},
+};
+
+/// The maximum length of a transaction index in a block in nibbles.
+/// Theoretically a block can have up to 1428 transactions in Ethereum, which takes 3 bytes to represent.
+const MAX_INDEX_NIBBLES: usize = 6;
+// Contract extraction public Inputs:
+/// - `H : [8]F` : packed node hash
+const H_RANGE: PublicInputRange = 0..PACKED_HASH_LEN;
+/// - `K : [6]F` : Length of the transaction index in nibbles
+const K_RANGE: PublicInputRange = H_RANGE.end..H_RANGE.end + MAX_INDEX_NIBBLES;
+/// `T : F` pointer in the MPT indicating portion of the key already traversed (from 6 → 0)
```

Comment:
6 should be a const generic right ? why 6 ?

Reply:
I was working off of the 30 million gas block limit on Ethereum. Since any transaction costs a minimum of 21 000 gas the theoretical maximum number of transactions is 1428 in a block which is 3 bytes long, so 6 nibbles.

Reply:
Correction its two bytes long so should only be 4 nibbles, I forgot how to do maths there for a little bit.

Reply:
Great, can you write this down in the source code then ? 🙏